gobblin-aws/src/main/java/gobblin/aws/AWSClusterSecurityManager.java

```diff
@@ -0,0 +1,158 @@
+package gobblin.aws;
+
+import java.io.IOException;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.amazonaws.auth.BasicAWSCredentials;
+import com.amazonaws.auth.BasicSessionCredentials;
+import com.amazonaws.services.securitytoken.AWSSecurityTokenServiceClient;
+import com.amazonaws.services.securitytoken.model.AssumeRoleRequest;
+import com.amazonaws.services.securitytoken.model.AssumeRoleResult;
+import com.google.common.base.Optional;
+import com.google.common.base.Throwables;
+import com.google.common.util.concurrent.AbstractIdleService;
+import com.typesafe.config.Config;
+
+import gobblin.annotation.Alpha;
+import gobblin.util.ExecutorsUtils;
+
+/**
+ * Class for managing AWS login and credentials renewal.
+ *
+ * <p>
+ *   This class makes use of {@link BasicAWSCredentials} and {@link BasicSessionCredentials} to
+ *   manage renewal of Amazon AWS authentication.
+ *   This class runs a scheduled login executor
+ *   task to refresh the credentials upon specified renewal interval which is configurable.
+ * </p>
+ *
+ * @author Abhishek Tiwari
+ */
+@Alpha
+public class AWSClusterSecurityManager extends AbstractIdleService {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(AWSClusterSecurityManager.class);
+
+  private final Config config;
+
+  private volatile String serviceAccessKey;
+  private volatile String serviceSecretKey;
+  private volatile boolean clientAssumeRole;
+  private volatile String clientRoleArn;
+  private volatile String clientExternalId;
+  private volatile String clientSessionId;
+  private volatile long lastRefreshTimeInMillis;
```

Comment:
can't these all be `final` rather than `volatile`, they all seem to be only set once and derive there values from a `Config` object which is immutable.